ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/TableTableJoinBuilder.java

```diff
@@ -52,6 +53,6 @@ private TableTableJoinBuilder() {
         throw new IllegalStateException("invalid join type");
     }
 
-    return left.withTable(result, joinParams.getSchema());
+    return left.withTable(result, joinParams.getSchema(), Optional.empty());
```

Comment:
Why do you need to explicitly avoid passing the materialization info builder here?  Just curious.  Is it because joins aren't materialized and the old code would have implied that they were?

You could probably do `KTableHolder.unmaterialized(...)` and pass in the ExecutionKeyFactory from left to make it a bit more explicit.

Reply:
>  Is it because joins aren't materialized and the old code would have implied that they were?

exactly this reason

Reply:
+1 on using `unmaterialized()` instead.